gateleen-hook/src/test/java/org/swisspush/gateleen/hook/HookHandlerTest.java

```diff
@@ -50,8 +60,23 @@
 
     private HookHandler hookHandler;
 
+    private static final Method registerListenerMethod;
+    private ListenerRepository listenerRepository;
+
+    static {
+        // Lever out access restrictions to make this a true unit test and not an integration
+        // test testing through hundreds of layers.
+        try{
+            registerListenerMethod = HookHandler.class.getDeclaredMethod( "registerListener" , Buffer.class );
```

Comment:
Don't test private methods like this. Just google the reasons why not :-)

A nice quote from the book "Pragmatic Unit Testing":

> In general, you don't want to break any encapsulation for the sake of testing (or as Mom used to say, "don't expose your privates!"). Most of the time, you should be able to test a class by exercising its public methods. If there is significant functionality that is hidden behind private or protected access, that might be a warning sign that there's another class in there struggling to get out.

meaning when you struggle to test the private methods through the public methods, it indicates a possible code smell which should be refactored.